File: spring-boot-ecommerce/src/main/java/com/digantasom/ecommerce/service/CheckoutService.java
package com.digantasom.ecommerce.service;

import com.digantasom.ecommerce.dto.PaymentInfo;
import com.digantasom.ecommerce.dto.Purchase;
import com.digantasom.ecommerce.dto.PurchaseResponse;
import com.stripe.exception.StripeException;
import com.stripe.model.PaymentIntent;

public interface CheckoutService {

  PurchaseResponse placeOrder(Purchase purchase);

  PaymentIntent createPaymentIntent(PaymentInfo paymentInfo) throws StripeException;
}

File: angular-ecommerce/src/app/services/checkout.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { Purchase } from '../common/purchase';
import { PaymentInfo } from '../common/payment-info';

import { PurchaseResponse } from '../common/purchase-response';

import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root',
})
export class CheckoutService {
  private purchaseUrl: string = `${environment.luv2shopApiUrl}/checkout/purchase`;
  private paymentIntentUrl: string = `${environment.luv2shopApiUrl}/checkout/payment-intent`;

  constructor(private http: HttpClient) {}

  placeOrder(purchase: Purchase): Observable<PurchaseResponse> {
    return this.http.post<PurchaseResponse>(this.purchaseUrl, purchase);
  }

  createPaymentIntent(paymentInfo: PaymentInfo): Observable<any> {
    return this.http.post<PaymentInfo>(this.paymentIntentUrl, paymentInfo);
  }
}
